diff -Naur lo-menubar-0.1.0/wscript lo-menubar-0.1.0.new/wscript
--- lo-menubar-0.1.0/wscript	2011-03-21 13:05:23.000000000 +0100
+++ lo-menubar-0.1.0.new/wscript	2011-09-26 16:59:30.568170115 +0200
@@ -16,7 +16,7 @@
 	               help='Libreoffice prefix')
    	opt.add_option('--ure-prefix',
 	               action='store',
-	               default='/usr/lib/ure',
+	               default='/usr/lib/libreoffice/ure',
 	               dest="URE_PREFIX",
 	               help='Libreoffice prefix')
 
@@ -24,21 +24,22 @@
 def configure(conf):
 	conf.load('compiler_cxx')
 	
-	uno_sdk_libpath = '%s/basis3.3/sdk/lib' % Options.options.LO_PREFIX
+	uno_sdk_libpath = '%s/basis-link/sdk/lib' % Options.options.LO_PREFIX
 	
 	conf.check_cxx(lib='uno_sal', uselib_store='SALLIB', libpath=uno_sdk_libpath, mandatory=True)
-	conf.check_cxx(lib='uno_cppu', uselib_store='CPPULIB', libpath=uno_sdk_libpath, mandatory=True)
-	#conf.check_cc(lib='uno_salhelpergcc3', uselib_store='SALHELPERLIB',  mandatory=True)
-	conf.check_cxx(lib='uno_cppuhelpergcc3', uselib_store='CPPUHELPERLIB', libpath=uno_sdk_libpath,  mandatory=True)
+	conf.check_cxx(lib='uno_cppu', uselib=['SALLIB'], uselib_store='CPPULIB', libpath=uno_sdk_libpath, mandatory=True)
+	conf.check_cc(lib='uno_salhelpergcc3', uselib_store='SALHELPERLIB',  mandatory=True)
+	conf.check_cxx(lib='uno_cppuhelpergcc3', uselib=['SALLIB', 'CPPULIB', 'SALHELPERLIB'],
+		       uselib_store='CPPUHELPERLIB', libpath=uno_sdk_libpath,  mandatory=True)
 	
-	conf.find_program('cppumaker', var='CPPUMAKER', path_list=[Options.options.LO_PREFIX + '/basis3.3/sdk/bin'], mandatory=True)
+	conf.find_program('cppumaker', var='CPPUMAKER', path_list=[Options.options.LO_PREFIX + '/basis-link/sdk/bin'], mandatory=True)
 
 	conf.env['TYPES_RDB'] = conf.find_file('types.rdb',
 	                                       path_list=[Options.options.URE_PREFIX + '/share/misc/'],
 	                                       mandatory=True)
 	
 	conf.env['OFFAPI_RDB'] = conf.find_file('offapi.rdb',
-	                                        path_list=[Options.options.LO_PREFIX + '/basis3.3/program'],
+	                                        path_list=[Options.options.LO_PREFIX + '/basis-link/program'],
 	                                        mandatory=True)
 	
 	conf.check_cfg(package='gio-2.0', uselib_store='GIO',
@@ -52,9 +53,11 @@
 	conf.check_cfg(package='gtk+-2.0', uselib_store='GTK',
 	               atleast_version='2.24.1', args='--cflags --libs')
 	conf.check_cfg(package='atk', uselib_store='ATK',
-	               atleast_version='1.33.6', args='--cflags --libs')
+	               atleast_version='1.32.0', args='--cflags --libs')
+	               #atleast_version='1.33.6', args='--cflags --libs')
 	conf.check_cfg(package='gdk-pixbuf-2.0', uselib_store='GDKPIXBUF',
-	               atleast_version='2.23.0', args='--cflags --libs')
+	               atleast_version='2.22.1', args='--cflags --libs')
+	               #atleast_version='2.23.0', args='--cflags --libs')
 	conf.check_cfg(package='gdk-2.0', uselib_store='GDK',
 	               atleast_version='2.24.1', args='--cflags --libs')	               
 	conf.check_cfg(package='x11', uselib_store='X11', args='--cflags --libs')
@@ -181,7 +184,7 @@
 	                  'src/MenuItemInfo.cxx',
 	                  'src/MenuItemStatusListener.cxx'],
 	          target=target,
-	          uselib=['SALLIB', 'CPPULIB', 'CPPUHELPERLIB', 'GIO', 'DBUSMENU', 'DBUSMENUGTK', 'GDKPIXBUF', 'GTK', 'GDK', 'ATK', 'X11', 'XPROTO'],
+	          uselib=['SALLIB', 'CPPULIB', 'SALHELPERLIB', 'CPPUHELPERLIB', 'GIO', 'DBUSMENU', 'DBUSMENUGTK', 'GDKPIXBUF', 'GTK', 'GDK', 'ATK', 'X11', 'XPROTO'],
 	          includes=includes,
 	          defines=['UNX', 'SAL_UNX', 'CPPU_ENV=gcc3'],
 	          install_path='%s/share/extensions/%s/%s' % (Options.options.LO_PREFIX, target, lo_platform),
@@ -202,7 +205,7 @@
 #	def post_install (ctx):
 #		if ctx.cmd != 'install':
 #			return
-#		ctx.exec_command ('/usr/lib/ure/bin/regcomp -register -wop="vnd.sun.star.expand:\$OOO_BASE_DIR/program/" -r /usr/lib/libreoffice/basis3.3/program/appmenu.rdb -c /usr/lib/libreoffice/basis3.3/program/%s.uno.so' % 'menubar')
+#		ctx.exec_command ('/usr/lib/ure/bin/regcomp -register -wop="vnd.sun.star.expand:\$OOO_BASE_DIR/program/" -r /usr/lib/libreoffice/basis-link/program/appmenu.rdb -c /usr/lib/libreoffice/basis-link/program/%s.uno.so' % 'menubar')
 #	
 #	bld.add_post_fun (post_install)
 
