diff --git a/projectmanagers/cmake/cmakemanager.cpp b/projectmanagers/cmake/cmakemanager.cpp
index 7fd75da..f0f9098 100644
--- a/projectmanagers/cmake/cmakemanager.cpp
+++ b/projectmanagers/cmake/cmakemanager.cpp
@@ -26,6 +26,7 @@
 #include <QDomDocument>
 #include <QDir>
 #include <QQueue>
+#include <QRegExp>
 
 #include <kpluginfactory.h>
 #include <kpluginloader.h>
@@ -650,14 +651,38 @@ KDevelop::ReferencedTopDUContext CMakeManager::includeScript(const QString& file
 
 QSet<QString> filterFiles(const QStringList& orig)
 {
+    QStringList filteredFiles = ICore::self()->projectController()->filteredFiles();
     QSet<QString> ret;
     foreach(const QString& str, orig)
     {
-        ///@todo This filter should be configurable, and filtering should be done on a manager-independent level
-        if (str.endsWith(QLatin1Char('~')) || str.endsWith(QLatin1String(".bak")))
-            continue;
+        ///@todo Filtering should be done on a manager-independent level
+        bool insert = true;
+
+        foreach(const QString& filter, filteredFiles)
+        {
+            if ( filter.contains(QLatin1String( "*" )) )
+            {
+                // Glob, so use regexp
+                QRegExp rx( filter );
+                rx.setPatternSyntax( QRegExp::Wildcard );
+                if ( rx.exactMatch( str ) )
+                {
+                    insert = false;
+                    break;
+                }
+            }
+            else
+            {
+                // This is not a regexp, so use startsWith + length
+                if ( str.startsWith( filter ) && (str.length() == filter.length()) )
+                {
+                    insert = false;
+                    break;
+                }
+            }
+        }
 
-        ret.insert(str);
+        if ( insert ) ret.insert(str);
     }
     return ret;
 }
